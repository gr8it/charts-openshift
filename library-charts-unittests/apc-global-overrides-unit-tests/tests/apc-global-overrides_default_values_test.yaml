suite: APC global overrides - default values
templates:
  - templates/apc-global-overrides.yaml
release:
  name: agout
  namespace: default

tests:
  - it: manifest should match snapshot
    asserts:
      - matchSnapshot: {}

  - it: customer from default values
    asserts:
      - equal: { path: data.customer, value: ~ }

  - it: repo fields from default values
    asserts:
      - equal: { path: data.repoURL, value: ~ }
      - equal: { path: data.repoTargetRevision, value: ~ }

  - it: environment fields from default values
    asserts:
      - equal: { path: data.environment, value: ~ }
      - equal: { path: data.environmentShort, value: ~ }

  - it: cluster identity fields from default values
    asserts:
      - equal: { path: data.clusterName, value: ~ }
      - equal: { path: data.clusterAcmName, value: ~ }
      - equal: { path: data.clusterType, value: ~ }

  - it: cluster domain and API fields from default values
    asserts:
      - equal: { path: data.clusterBaseDomain, value: ~ }
      - equal: { path: data.clusterAppsDomain, value: ~ }
      - equal: { path: data.clusterApiURL, value: ~ }

  - it: cluster API versions from default values
    asserts:
      - equal: { path: data.clusterKubeVersion, value: ~ }
      - equal:
          path: data.clusterApiVersions
          value: []

  - it: clusterServices and merged default values
    asserts:
      - equal:
          path: data.clusterServices
          value: {}
      - equal:
          path: data.clusterServicesMerged
          value: {}

  - it: booleans from default values
    asserts:
      - equal: { path: data.clusterIsHub, value: false }
      - equal: { path: data.clusterRunsApps, value: false }

  - it: proxy, noProxy, proxyIPs from default values
    asserts:
      - equal: { path: data.proxy, value: ~ }
      - equal: { path: data.noProxy, value: ~ }
      - equal:
          path: data.proxyIPs
          value: []

  - it: services and merged default values
    asserts:
      - equal:
          path: data.services
          value: {}
      - equal:
          path: data.servicesMerged
          value: {}

  - it: caCertificates and merged default values
    asserts:
      - equal:
          path: data.caCertificates
          value: {}
