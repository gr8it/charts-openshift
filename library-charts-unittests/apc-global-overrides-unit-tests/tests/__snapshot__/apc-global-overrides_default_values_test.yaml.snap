manifest should match snapshot:
  1: |
    data:
      clusterAcmName: null
      clusterApiURL: null
      clusterApiVersions: []
      clusterAppsDomain: null
      clusterBaseDomain: null
      clusterIsHub: false
      clusterKubeVersion: null
      clusterName: null
      clusterRunsApps: false
      clusterServices: {}
      clusterServicesExtracted: null
      clusterServicesMerged: {}
      clusterType: null
      customer: null
      environment: null
      environmentShort: null
      repoTargetRevision: HEAD
      repoURL: null
      services: {}
      servicesExtracted:
        certManager:
          defaultClusterIssuer: null
        crossplane:
          kubeVaultProviderConfigName: null
        externalSecretsOperator:
          defaultClusterSecretStore: null
        vault:
          kubeAuthMountPath: null
          name: null
          url: null
      servicesMerged: {}
